### ISSUE TITLE
Potential coding error - existing temptables found when disposing database. Must be dropped!'

### ISSUE THREAD

**Issue by deedy-tech** [Syxton/moodle-tool_coursearchiver#63]
I am testing the Course Archiver on a Moodle platform that has a lot of courses going back a number of years. When I bulk archive courses, sometimes it works and sometimes I am getting an error 
![Screenshot 2024-09-25 at 15 35 34](https://github.com/user-attachments/assets/d2b1fa05-5d9c-4884-a6d5-cd045b48c768)

In the error log I am getting the following errors
Got error 'PHP message: Exception ignored in shutdown function tool_log\\log\\manager::dispose: Error reading from database; PHP message: Exception ignored in shutdown function tool_monitor\\eventobservers::process_buffer: Error reading from database', referer: https://XXXXXX.com/admin/tool/coursearchiver/step4.php

Got error '; PHP message: Potential coding error - existing temptables found when disposing database. Must be dropped!', referer: https://XXXXXX.com/admin/tool/coursearchiver/step4.php

**Comment by Syxton** [comment]
@deedy-tech Can you navigate to one of the failed courses and perform a full backup?  If the backup fails there is something wrong with either backup or the data in the course.  My plugin only really manages and commands for backups to occur.  The code that does the backup is core Moodle.  Could you check to see if the standard backup fails?

**Comment by deedy-tech** [comment]
Thanks for the reply, you were right the problem was with the backup code. I ran the following with DEBUG turned on 

php admin/cli/backup.php  --destination=/backup/ --courseid=90

It first showed up an issue in /backup/util/xml/xml_writer.class.php, the xml_safe_text_content function needed to be hotfixed to accept an array as argument and implode() it. (https://moodle.org/mod/forum/discuss.php?d=452695#p1819612)

I then got an 'MySQL server has gone away' and fixed that by increasing the MYSQL wait_timeout variable.

It is now working correctly.
